src/cellfinder_core/detect/detect.py

```diff
@@ -37,26 +38,26 @@ def calculate_parameters_in_pixels(
 
 
 def main(
-    signal_array,
-    start_plane,
-    end_plane,
-    voxel_sizes,
-    soma_diameter,
-    max_cluster_size,
-    ball_xy_size,
-    ball_z_size,
-    ball_overlap_fraction,
-    soma_spread_factor,
-    n_free_cpus,
-    log_sigma_size,
-    n_sds_above_mean_thresh,
-    outlier_keep=False,
-    artifact_keep=False,
-    save_planes=False,
-    plane_directory=None,
+    signal_array: np.ndarray,
+    start_plane: int,
+    end_plane: int,
+    voxel_sizes: Tuple[float, float, float],
```

Comment:
I need to check whether e.g. cellfinder passes a tuple or a list.

Reply:
The motivation for choosing `Tuple` instead of `List` here was because tuples can be fixed length, but lists can't. Given type checking isn't runtime this won't stop anyone passing a list of floats (ie it won't break any existing code), but I think it's good to indicate that this should be three values here?

Happy to switch to `Sequence[int]` which would allow a list or a tuple here though.

Reply:
Nah you're right, I think `Tuple` is correct. I forgot the typing doesn't affect runtime.